ravedude/src/main.rs

```diff
@@ -72,7 +79,9 @@ struct Args {
     /// * nano168
     /// * duemilanove
     #[structopt(name = "BOARD", verbatim_doc_comment)]
-    board: String,
+    // When Ravedude.toml exists, the binary is placed where the board should be. This is an OsString to not lose
+    // informaton when we have to take the board as the binary.
+    board: Option<OsString>,
```

Comment:
Got a bit of a tricky problem here right now:  The `ravedude --help` output is still completely wrong.  Ideally, we can get it to just show the new way of doing things and secretly handle the legacy method in the background.

To do this, I think we need to take the following steps:

1. Rename the current `bin` to something like `oldbin` with the help message stating that this is an argument used for backwards compatibility.
2. Rename the current `board` to `bin`.  Adjust the docstring to the current one from `bin` and drop the `verbatim_doc_comment` setting.

Reply:
ravedude's manifest format is already a bit complex. I'd say a full description of the manifest keys and what they do would be too long to be directly included in the `ravedude --help` output, so I have a few ideas:

1. Add another command (E.g. `ravedude --help-toml` or the like) which prints out documentation for the manifest format. This would very simply and effectively get the job done but might be hard to scroll/search/etc.
2. Use mdBook or the like to generate online documentation for ravedude. Apart from being harder to setup (especially for offline viewing), it's probably out of scope for this PR. I think this would be the best solution long-term, though, especially if we can write and maintain docs for the rest of avr-hal as well. Or we could just link to GitHub's built-in Markdown viewer/a GitHub wiki.

Regardless, I'm going to work on some basic documentation for the manifest format. I think Option 1 is the best for now.

Reply:
another option would be writing/generating a `ravedude.5` man page (5 is for file formats).

The advantage is that this is available online (either the github markdown renderer or manpage sites) and offline and is searchable.

A manpage probably can’t be shipped with cargo install, but it can be shipped by packaging ravedude. (why not?)

Reply:
Sorry for the long wait...  I agree that documentation is out of scope for this PR, let's take care of that in the next step.  My original comment was mostly about fixing the help generated from structopt which you have taken care of, thanks a lot.

For documentation I created issue #579 to track it.